packages/wonder-stuff-testing/src/jest/after-each-restore-env.ts

```diff
@@ -0,0 +1,56 @@
+import {afterEach} from "./internal/jest-wrappers";
+
+/**
+ * Restore the values of the given environment variables after each test.
+ *
+ * This captures the values of the given environment variables on invocation
+ * and then, after each test case, it restores those values.
+ *
+ * @param variableNames The names of the environment variables to restore.
+ */
+export const afterEachRestoreEnv = (
+    ...variableNames: ReadonlyArray<string>
+): void => {
+    // We capture the variables to restore. If none are given, we capture all.
+    const restoreAll = variableNames.length === 0;
+    const variablesToCapture = restoreAll
+        ? Object.keys(process.env)
+        : variableNames;
+    // We capture the current values on invocation.
+    const originalValues = variablesToCapture.reduce(
```

Comment:
Does this have to be in a `beforeEach`? So that the values match what they were before the test began to run - not when it was initialized?

Reply:
I don't think so. In all the cases this is done in code, we snapshot the values at the start of the `describe`. The `beforeEach` would capture potentially mutated state and therefore allow for bleeding of changes across test cases, which we don't want.